### PR TITLE
Handle operators adjacent to @@variables

### PR DESCRIPTION
**Pull request by robverschoor**
### Description

Operators which are directly adjacent to `@@` variables were incorrectly getting interpreted as PG operators. A space is now inserted in ANTLR prior to `@@` to ensure they are parsed correctly.
In addition, parameter defaults that use `@@` variables (`create proc p @p int=@@spid as ...`) and named parameter syntax (`exec myproc @p=@@spid`) are rewritten in the same way.

Signed-off-by: Rob Verschoor [rcv@amazon.com](mailto:rcv@amazon.com)

### Issues Resolved

BABEL-2888 =@@, >@@ (etc) cause syntax error 

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** Yes

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/upgrade/latest/schedule [hunk 1]
```diff
@@ -466,6 +466,7 @@ print_null
 unquoted_string
 doublequoted_string
 operator_whitespace
+operator_atatvar
```
Comment (forestkeeper) [inline test/JDBC/upgrade/latest/schedule:469]: For this Mvu test, we should put it into different upgrade test files

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -2130,6 +2130,63 @@ class tsqlBuilder : public tsqlCommonMutator
...
+
+		if ((ctx->comparison_operator()) && (ctx->expression().size() > 1))
+		{
+			std::string op = getFullText(ctx->comparison_operator());		
+			if ((op.back() == '=') || 
+			    (op.back() == '>') || 
+			    (op.back() == '<'))
+			{
+				// The operator must be followed immediately by the variable without any character in between
+				Assert(ctx->expression(1));
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2148]: Would you mind assert for 
ctx->expression(1)->start
ctx->expression(1)->stop
 too ?

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 3]
```diff
@@ -2130,6 +2130,63 @@ class tsqlBuilder : public tsqlCommonMutator
...
+				size_t startPosition = ctx->expression(1)->start->getStartIndex();
+				if ((startPosition - ctx->comparison_operator()->stop->getStopIndex()) == 1)
+				{
+					std::string var = getFullText(ctx->expression(1));
+					// The subsequent expression must be a variable starting with '@@'
+					if (var.front() == '@') 
+					{
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2155]: Those two if condition can be combine into one

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 4]
```diff
@@ -2130,6 +2130,63 @@ class tsqlBuilder : public tsqlCommonMutator
...
+		Assert(ctx->execute_parameter());
+		size_t startPosition = ctx->execute_parameter()->start->getStartIndex();
+		if ((startPosition - ctx->EQUAL()->getSymbol()->getStopIndex()) == 1)
+		{
+			std::string var = getFullText(ctx->execute_parameter());
+				
+			// The subsequent expression must be a variable starting with '@@'
+			if (var.front() == '@') 
+			{
+				if (var.at(1) == '@')
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2180]: Those three if condition can combine into one.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 5]
```diff
@@ -2709,6 +2766,27 @@ class tsqlMutator : public TSqlParserBaseListener
...
+		{
+			// The '=' char must be followed immediately by the variable without any character in between
+			Assert(ctx->expression());			
+			size_t startPosition = ctx->expression()->start->getStartIndex();
+			if ((startPosition - ctx->EQUAL()->getSymbol()->getStopIndex()) == 1)
+			{
+				std::string var = getFullText(ctx->expression());
+				// The subsequent default expression must be a variable starting with '@@'
+				if (var.front() == '@') 
+				{
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2781]: Those two if condition can be combined into one

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 2 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2148]: start and stop always have a value for a node/token. if we cannot rely on that, we cannot rely on ANTLR

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 3 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2155]: Indeed they can, but I am following the principle that the second test is required only when the first succeeds, so don't do the test unless it is truly required.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 4 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2180]: see comment above

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 5 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2781]: see comment above

File: test/JDBC/upgrade/latest/schedule (hunk 1 above)
Comment (robverschoor) [reply test/JDBC/upgrade/latest/schedule:469]: there is no upgrade aspect since the things being fixed here all throw errors prior to the fix

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 2 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2148]: To illustrate, start and stop are not explicitly checked for  a non-null value anywhere in the code.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 3 above)
Comment (forestkeeper) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2155]: The runtime will automatically do this so won't be a problem. I suggested this since that'll help reading the code.

**Review by forestkeeper — APPROVED** [review]
